package.yaml

```diff
@@ -58,3 +58,5 @@ tests:
       - hspec
       - shellwords
       - megaparsec
+    build-tools:
+      - hspec-discover:hspec-discover
```

Comment:
I'm curious if you can explain why this is necessary for you. I've never added this anywhere and my test suites have always worked fine, in all my projects locally and on CI.

Reply:
Do you have an `hspec-discover` binary in your `$PATH`? Maybe you `cabal install`ed it at some point. I'm not sure. I'm also not using Stack, if that makes a difference.

Reply:
Not normally,

```console
% which hspec-discover
hspec-discover not found
```

But in the context of the project, the dependencies' executables are present:

```console
% stack exec which hspec-discover
/home/patrick/.stack/snapshots/x86_64-linux-tinfo6/cb3005caf1c362da0e393ac80f1e4826f5e17c857020e4eda5dd9aad47727a7a/9.6.6/bin/hspec-discover
``

This makes sense because `hspec-discover` is a dependency in the `test` target.

```console
% tail package.yaml
    main: Spec.hs
    source-dirs: test
    ghc-options:
      - -threaded
      - -rtsopts
      - -with-rtsopts=-N
    dependencies:
      - hspec
      - shellwords
      - megaparsec
```

And, without any new `build-tools`, it still works fine:

```console
% stack test
shellwords> test (suite: hspec)


ShellWords
  parser
    parses |var --bar=baz| correctly [✔]
    parses |var --bar="baz"| correctly [✔]
    ...
```

Reply:
Ahh, that explains it! `cabal run hspec-discover` works too, but the `hspec-discover` binary isn't inserted into the `$PATH` when building/running tests.

Reply:
Even if not using stack, wouldn't you want to run something like `cabal test` or `cabal build --test`, or even `cabal run hspec`? I'm a little confused as to why you'd ever run `hspec-discover` directly as that's a source plugin meant to generate a module.

Reply:
That's correct, `cabal test` does not work!

Reply:
I included the error in the commit message:

```
$ cabal test
Build profile: -w ghc-9.6.6 -O1
In order, the following will be built (use -v for more details):
 - shellwords-0.1.3.2 (test:hspec) (first run)
Preprocessing test suite 'hspec' for shellwords-0.1.3.2...
Building test suite 'hspec' for shellwords-0.1.3.2...
ghc-9.6.6: could not execute: hspec-discover
```

Reply:
:exploding_head: wow, that's surprising!

Alright, well, I have to run for the weekend, so won't get to merge/release this for a bit anyway, but if I can't figure out more I'll just include it.

Reply:
Sure. I'll split it into a separate PR at least.